### ISSUE TITLE
Global install failures on ATDM waterman builds

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#5834]
CC: @fryeguy52 

## Description

As shown in [this query](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&date=2019-08-31&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=61&value1=Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt&field2=buildstarttime&compare2=83&value2=2019-08-01) the installs of the build `Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt` used by SPARC Trilinos Integration Testing started on 8/27/2019 and show the error ([here](https://testing-dev.sandia.gov/cdash-dev-view/viewBuildError.php?buildid=4814866)) for the target `install_package_by_package` showing:

```
CMake Error at cmake_install.cmake:90 (file):
  file INSTALL cannot copy file
  "/home/jenkins/waterman/workspace/Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt/Trilinos/cmake/std/atdm/waterman"
  to
  "/home/atdm-devops-admin/trilinos_installs/2019-08-27/waterman_cuda-9.2_fpic_static_opt/share/atdm-trilinos/waterman".



***
*** The global install failed so resorting to package-by-package installs
***


CMake Error at cmake_pbp_install.cmake:25 (message):
  Error, full install failed!
```

The commits pulled that day are shown [here](https://testing-dev.sandia.gov/cdash-dev-view/viewNotes.php?buildid=4814866#!#note2) and show the likely breaking commit:

```
fbec823:  Fix load_matching_env.sh script for custom envs (#4933)
Author: Roscoe A. Bartlett <rabartl@sandia.gov>
Date:   Mon Aug 26 14:57:25 2019 -0600

M	cmake/std/atdm/ATDMDevEnvSettings.cmake
```

**Comment by bartlettroscoe** [comment]
I tried to reproduce this locally in my account 'rabartl' and my first attempt I could not do so. 

Using the Trilinos version:

```
95a8f30 "Merge pull request #5828 from cgcgcg/miniEMfix"
Author: Christian Glusa <cgcgcg@users.noreply.github.com>
Date:   Fri Aug 30 13:33:43 2019 -0600 (27 hours ago)
```

I did:

```
$ cd ~/Trilinos.base/BUILDS/WATERMAN/CHECKIN/waterman_cuda-9.2_fpic_static_opt/

$ source /home/rabartl/Trilinos.base/Trilinos/cmake/std/atdm/load-env.sh \
    waterman_cuda-9.2_fpic_static_opt
Hostname 'waterman11' matches known ATDM host 'waterman' and system 'waterman'
Setting compiler and build options for build-name 'waterman_cuda-9.2_fpic_static_opt'
Using waterman compiler stack CUDA-9.2 to build RELEASE code with Kokkos node type CUDA and KOKKOS_ARCH=Power9,Volta70

$ rm -r CMake* packages cmake

$ time ./do-configure.base \
  -DTrilinos_ENABLE_Kokkos:BOOL=ON \
  -DTrilinos_ENABLE_ALL_OPTIONAL_PACKAGES:BOOL=ON \
  -DTrilinos_ENABLE_ALL_FORWARD_DEP_PACKAGES:BOOL=OFF \
   &> configure.out

real    0m49.563s
user    0m37.929s
sys     0m12.634s

$ time ninja -j20 &> make.out

real    3m45.746s
user    58m46.818s
sys     3m38.099s

$ time ninja -j20 install_package_by_package

real    0m3.799s
user    0m0.112s
sys     0m0.606s
```

I wonder if this has something to do with that fact that the Jenkins job runs the install using the command run-as-atdm-devops-admin.  That is what I will try next.

**Comment by bartlettroscoe** [comment]
Wait, I know what problem is.  The problem is that the 'install' command is trying to copy a file from:

```
/home/jenkins/waterman/workspace/Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt/Trilinos/cmake/std/atdm/waterman
```

run as the 'atdm-devops-admin' account through the 'run-as-atdm-devops-admin' program.  Well the 'atdm-devops-admin' user can't read anything under the directory `/home/jenkins/`.  It can only read things under the directory `/home/atdm-devops-admin/` where the configure command is actually run.  The problem is that the original env is loaded from the script:

```
/home/jenkins/waterman/workspace/Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt/Trilinos/cmake/std/atdm/waterman
```

which set the env var `ATDM_CONFIG_SYSTEM_DIR` before the new directory:

```
/home/atdm-devops-admin/jenkins/waterman_cuda-9.2_fpic_static_opt/
```
was created and cd'ed into.  We can see this in the output:

* https://jenkins-srn.sandia.gov/view/Trilinos%20ATDM/job/Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt/117/consoleFull

which shows:

```
ATDM_CONFIG_SYSTEM_DIR=/home/jenkins/waterman/workspace/Trilinos-atdm-waterman_cuda-9.2_fpic_static_opt/Trilinos/cmake/std/atdm/waterman
```

That is bad that that is what is causing this.

I can fix this in one of two ways.  One approach is the create and cd into the new working directory under /home/atdm-devops-admin/ before sourcing the 'load-env.sh' script.  But info on what directory that should be is set in the `ATDM_CONFIG_WORKSPACE_BASE` in 'atdm/load-env.sh' from the 'atdm/waterman/enviornment.sh' script setting `ATDM_CONFIG_WORKSPACE_BASE_DEFAULT`.  So we can't create and cd into the new working directory before sourcing the 'load-env.sh' script.

The second approach is to change the code in 'atdm/ATDMDevOpsSettings.cmake' to instead of installing from `ATDM_CONFIG_SYSTEM_DIR` for an offical configuration to instead install from `${CMAKE_CURRENT_LIST_DIR}/${ATDM_CONFIG_SYSTEM_NAME}`.  That should be an easy change to make.

**Comment by bartlettroscoe** [comment]
The fix should be in PR #5835.  I manually merged the branch into 'atdm-nightly' in the commit 2fe10ba so this will run tomorrow and hopefully fix the install of the Trilinos 'waterman' build (and therefore hopefully fix the SPARC Trilinos Integration 'waterman' build).

**Comment by bartlettroscoe** [comment]
As shown in [this query](https://testing-dev.sandia.gov/cdash-dev-view/index.php?project=Trilinos&date=2019-09-03&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-waterman&field2=buildstarttime&compare2=83&value2=2019-08-01), these install failures went away starting 9/3/2019.  In fact, I even looked at the SPARC Trilinos Integration build on 'waterman' for SPARC testing day 2019-09-05 and the build passed.

Therefore, closing as fixed.
